Engine/ac/dynobj/cc_audiochannel.cpp

```diff
@@ -16,7 +16,7 @@
 #include "media/audio/audiodefines.h"
 #include "ac/dynobj/scriptaudiochannel.h"
 
-extern ScriptAudioChannel scrAudioChannel[MAX_SOUND_CHANNELS + 1];
+extern ScriptAudioChannel scrAudioChannel[MAX_SOUND_CHANNELS];
 
```

Comment:
This may break scripts, but earlier probably will crash engine at InitAndRegisterAudioObjects() where it still iterates from 0 to MAX_SOUND_CHANNELS **inclusive**.

I think this comment about `+ 1 just for safety on array iterations` could have tricked you.
MAX_SOUND_CHANNELS is a number of channels the user may reserve for their custom audio types, and +1 is a special crossfade channel. Maybe there's a sense to introduce new constant that would be equal to MAX_SOUND_CHANNELS + 1 and use here.

Reply:
This was an accident. I didn't mean to change any array lengths in this commit. Ignore this, I will revert it